packages/sdk/src/HeadlessUniversalCheckout/Managers/RedirectManager.ts

```diff
@@ -0,0 +1,213 @@
+
+const eventTypes: EventType[] = [
+  'onRetrieved',
+  'onRetrieving',
+  'onInvalid',
+  'onError',
+  'onValid',
+  'onValidating'
+];
+
+export interface RedirectManagerProps {
+  paymentMethodType: string;
+  onRetrieved?: (metadata: any) => void;
+  onRetrieving?: () => void;
+  onError?: (errors: PrimerError[] | undefined) => void;
+  onInvalid?: (isValid: any) => void;
+  onValid?: () => void;
+  onValidating?: () => void;
+}
+
+class PrimerHeadlessUniversalCheckoutComponentWithRedirectManager {
+  options?: RedirectManagerProps;
+
+  ///////////////////////////////////////////
+  // Init
+  ///////////////////////////////////////////
+  constructor() { }
+
+  ///////////////////////////////////////////
+  // API
+  ///////////////////////////////////////////
+
+  async configure(
+    options: RedirectManagerProps
+  ): Promise<{ initializationData: PrimerInitializationData } | void> {
+    return new Promise(async (resolve, reject) => {
+      try {
+        this.options = options;
+        await this.configureListeners();
+
+        const data =
+          await RNTPrimerHeadlessUniversalCheckoutComponentWithRedirectManager.configure(
+            options.paymentMethodType
+          );
+
+        if (data) {
+          resolve(data);
+        } else {
+          resolve();
+        }
+      } catch (err) {
+        reject(err);
+      }
+    });
+  }
+
+  async submit(): Promise<{ initializationData: PrimerInitializationData } | void> {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const data =
+          await RNTPrimerHeadlessUniversalCheckoutComponentWithRedirectManager.submit();
+
+        if (data) {
+          resolve(data);
+        } else {
+          resolve();
+        }
+      } catch (err) {
+        reject(err);
+      }
+    });
+  }
+
+  async onBankSelected(
+    bankId: string
+  ): Promise<{ initializationData: PrimerInitializationData } | void> {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const data =
+          await RNTPrimerHeadlessUniversalCheckoutComponentWithRedirectManager.onBankSelected(
+            bankId
+          );
+
+        if (data) {
+          resolve(data);
+        } else {
+          resolve();
+        }
+      } catch (err) {
+        reject(err);
+      }
+    });
+  }
+
+  async onBankFilterChange(
+    filter: string
+  ): Promise<{ initializationData: PrimerInitializationData } | void> {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const data =
+          await RNTPrimerHeadlessUniversalCheckoutComponentWithRedirectManager.onBankFilterChange(
+            filter
+          );
+
+        if (data) {
+          resolve(data);
+        } else {
+          resolve();
+        }
+      } catch (err) {
+        reject(err);
+      }
+    });
+  }
+
+  async configureListeners(): Promise<void> {
+    //@ts-ignore
+    return new Promise(async (resolve, reject) => {
```

Comment:
💅🏻 [nitpick] the `Promise` constructor is unnecessary, simply marking the function `async` makes it return a promise already, e.g.:

```ts
async configureListeners(): Promise<void> {
  if (this.options?.onRetrieved)
    this.addListener('onRetrieved') // ...

  // ...
}

async example() {
  await this.configureListeners();
}
```